tsdb/series_partition.go

```diff
@@ -573,94 +575,103 @@ func (c *SeriesPartitionCompactor) Compact(p *SeriesPartition) error {
 	return nil
 }
 
-func (c *SeriesPartitionCompactor) compactIndexTo(index *SeriesIndex, seriesN uint64, segments []*SeriesSegment, path string) error {
-	hdr := NewSeriesIndexHeader()
-	hdr.Count = seriesN
-	hdr.Capacity = pow2((int64(hdr.Count) * 100) / SeriesIndexLoadFactor)
-
-	// Allocate space for maps.
-	keyIDMap := make([]byte, (hdr.Capacity * SeriesIndexElemSize))
-	idOffsetMap := make([]byte, (hdr.Capacity * SeriesIndexElemSize))
+var errDone error = errors.New("done")
 
-	// Reindex all partitions.
-	var entryN int
-	for _, segment := range segments {
-		errDone := errors.New("done")
+func (c *SeriesPartitionCompactor) compactIndexTo(index *SeriesIndex, seriesN uint64, segments []*SeriesSegment, path string) (err error) {
 
-		if err := segment.ForEachEntry(func(flag uint8, id uint64, offset int64, key []byte) error {
+	hdr := NewSeriesIndexHeader()
+	var keyIDMap []byte
+	var idOffsetMap []byte
+
+	hdr.Count = math.MaxUint64
+	seriesCount := seriesN
+	for {
+		seriesN = seriesCount
```

Comment:
I'm a little confused by what's going on here. Part of this may be due to changing the value of the input parameter `seriesN`. This changes the meaning of `seriesN` in the function call vs the function body. Copying to a different variable before using might help. A comment explaining what `seriesN` and `seriesCount` represent as the function runs would also help.

Reply:
That helps. Weird whitespace, but not going to worry about that.